server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -98,6 +128,38 @@ public void startReplication(final SegmentReplicationTarget target) {
         threadPool.generic().execute(new ReplicationRunner(replicationId));
     }
 
+    /**
+     * Checks if checkpoint should be processed
+     *
+     * @param requestCheckpoint       received checkpoint that is checked for processing
+     * @param indexShard      replica shard on which checkpoint is received
+     * @return true if checkpoint should be processed
+     */
+    private boolean shouldProcessCheckpoint(ReplicationCheckpoint requestCheckpoint, IndexShard indexShard) {
+        if (indexShard.state().equals(IndexShardState.STARTED) == false) {
```

Comment:
For my understanding what happens to a shard in RECOVERING state?

Reply:
We will take care of RECOVERING state logic in our next PR. Shards in RECOVERING state will never receive these new checkpoints.  We'll be updating the recovery process to first ping the primary to have the local shard tracked, and then force a replication event